src/widgets/mainwindow.cpp

```diff
@@ -664,25 +666,54 @@ void MainWindow::repairCacheButtonClicked()
     emit repairCache();
 }
 
+void MainWindow::stopNukeCountdown()
+{
+    _purgeTimer.stop();
+    updateStatusMessage(tr("Purge cancelled."));
+}
+
 void MainWindow::purgeArchivesButtonClicked()
 {
     if(_purgeTimer.isActive())
     {
-        _purgeTimer.stop();
-        _ui->purgeArchivesButton->setText(tr("Purge archives"));
-        _ui->purgeArchivesButton->setToolTip(tr("Nukes all archives"));
+        stopNukeCountdown();
     }
     else
     {
-        QMessageBox::StandardButton confirm = QMessageBox::question(this, tr("Confirm action")
-                                                                    ,tr("Are you sure you want to nuke all archives stored for this key?\nWarning: This action cannot be undone. All archives will be lost forever.")
-                                                                    ,( QMessageBox::Yes | QMessageBox::No ), QMessageBox::No);
-        if(confirm == QMessageBox::Yes)
+        const QString confirmationText = "No Tomorrow";
+        bool ok;
+        QString userText = QInputDialog::getText(this,
+            tr("Delete all archives?"),
+            tr("If you want to <b>delete all archives</b> stored for this key, type '%1'.<br><br>Warning: This action cannot be undone.  All archives will be lost forever.").arg(confirmationText),
+            QLineEdit::Normal,
+            tr("To confirm, type: %1").arg(confirmationText),
```

Comment:
Remove the text in the QLineEdit, since it makes it easy for users to proceed carelessly by selecting text and deleting instead of typing the words themselves.